include/boost/astronomy/coordinate/coord_sys/equatorial_coord.hpp

```diff
@@ -0,0 +1,194 @@
+//                double convert_to_ha()
+//                {
+//
+//                }
+            };
+
+            //Hour Angle
+            template<
+                    typename CoordinateType = double,
+                    typename HourAngleQuantity = bu::quantity<bu::si::plane_angle, CoordinateType>
+            >
+            struct HourAngle {
+            private:
+                HourAngleQuantity ha;
+            public:
+                HourAngle(){
+                    ha = 0.0 * bu::si::radian;
+                }
+
+                HourAngle(HourAngleQuantity _ha) : ha(_ha) {}
+
+                HourAngleQuantity get_angle() const{
+                    return static_cast<HourAngleQuantity>(ha);
+                }
+
+                void print() {
+                    std::cout << "Hour Angle: " << ha;
+                }
+
+//                //TODO: Use SOFA to convert
+//                double convert_to_ra()
+//                {
+//
+//                }
+            };
+
+            //Equatorial Coordinate Struct
+            template
+                    <
+                            typename CoordinateType = double,
+                            typename XQuantity = RightAscension<CoordinateType,bu::quantity<bu::si::plane_angle, CoordinateType>>,
+                            typename DeclinationQuantity = bu::quantity<bu::si::plane_angle, CoordinateType>
+                    >
+            struct equatorial_coord : public coord_sys
+                    <2, bg::cs::spherical<bg::degree>, CoordinateType> {
+            public:
+                //Default constructor
+                equatorial_coord() {}
+
+                equatorial_coord
+                        (
+                                XQuantity const &X,
+                                DeclinationQuantity const &Declination
+                        ) {
+                    this->set_X_Declination(X, Declination);
+                }
+
+                //Set X
+                void set_X(XQuantity const &X) {
+                    bg::set<0>(
+                            this->point,
+                            (X.get_angle()).value()
+                    );
+                }
+
+                //Set Declination
+                void set_Declination(DeclinationQuantity const &Declination) {
+                    bg::set<1>
+                            (
+                                    this->point,
+                                    static_cast<bu::quantity<bu::si::plane_angle, CoordinateType>>(Declination).value()
+                            );
+                }
+
+                //Set value of X and Declination
+                void set_X_Declination
+                        (
+                                XQuantity const &X,
+                                DeclinationQuantity const &Declination
+                        ) {
+                    this->set_X(X);
+                    this->set_Declination(Declination);
```

Comment:
we prefer `snake_case` over `CamelCase` this is quite confusing to use both in same. Make this change to every place needed

Reply:
Here I was using a combination of both.
For one word do you prefer CamelCase?
`typename EclipticLonQuantity = bu::quantity<bu::si::plane_angle, CoordinateType>,`
or
`typename ecliptic_lon_quantity = bu::quantity<bu::si::plane_angle, CoordinateType>,`

Will update functions like.
`this->set_EclipticLon_EclipticLat(EclipticLon, EclipticLat);`
is more readable over
`this->set_ecliptic_lon_ecliptic_lat(EclipticLon, EclipticLat);`

Reply:
1. Converted to `snake_case` 🐍
2. Following the convention of Latitude before Longitude everywhere.
3. In Equatorial Coordinate System I am using Latitude as first parameter and Declination (instead of Longitude) as second parameter.
4. Added description of every coordinate system and it’s parameters.
5. Refactored tests accordingly.